docs/nnx/nnx_basics.md

```diff
@@ -252,16 +254,86 @@ print(f'{y.shape = }')
 nnx.display(model)
 ```
 
-How do NNX transforms achieve this? To understand how NNX objects interact with
-JAX transforms lets take a look at the Functional API.
+## Using Modules as Pytrees
```

Comment:
I just realized pytree is a JAX concept, not a wider-spread python concept. 
Let's add this link when we talk about pytree: https://jax.readthedocs.io/en/latest/pytrees.html